### PR TITLE
Draw on map

### PR DESCRIPTION
**Pull request by shiloDrot**

### CODE REVIEW

File: demo/app/app.component.ts [hunk 1]
```diff
@@ -10,15 +11,52 @@ import { MapLayerProviderOptions } from '../../src/models/map-layer-provider-opt
...
-	arcGisMapServerProvider = MapLayerProviderOptions.ArcGisMapServer;
+	private arcGisMapServerProvider: MapLayerProviderOptions;
 	flyToOptions = {
 		duration: 2,
 		destination: Cesium.Cartesian3.fromDegrees(-117.16, 32.71, 15000.0),
 	};
+	private position: any;
+	private positions: any[] = new Array(2);
+	private redMatirial: any;
+	private aquamarine: any;
```
Comment (eitanfr) [inline demo/app/app.component.ts]: please resolve conflicts

File: src/components/ac-map/ac-map.component.ts [hunk 2]
```diff
@@ -19,7 +22,8 @@ import { PlonterService } from '../../services/plonter/plonter.service';
 @Component({
 	selector: 'ac-map',
 	template: '<ng-content></ng-content>',
-	providers: [CesiumService, BillboardDrawerService, CesiumEventBuilder, MapEventsManagerService, PlonterService]
+	providers: [CesiumService, BillboardDrawerService, CesiumEventBuilder, MapEventsManagerService, PlonterService,
+	LabelDrawerService, DynamicPolylineDrawerService, DynamicEllipseDrawerService]
```
Comment (shiloDrot) [inline src/components/ac-map/ac-map.component.ts]: I will add the other component in other PR.

File: src/services/basic-primitive-on-map/basic-primitive-on-map.service.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	removeFromMap() {
+		this.selfPrimitiveIsDraw = false;
+		return this._drawer.remove(this.selfPrimitive);
+	}
+
+	updateOnMap() {
+		if (this.selfPrimitiveIsDraw) {
+			this.removeFromMap();
```
Comment (shiloDrot) [inline src/services/basic-primitive-on-map/basic-primitive-on-map.service.ts]: I can't use it cause its not fit to this kind of drawing.
Its better to remove and add.
It doesn't make a performance issues cause we talk about only few objects

File: demo/app/app.component.ts [hunk 4]
```diff
@@ -10,15 +11,52 @@ import { MapLayerProviderOptions } from '../../src/models/map-layer-provider-opt
...
+
+	@ViewChild(AcLabelComponent) label: AcLabelComponent;
 
 	constructor() {
 	}
 
 	ngOnInit() {
+		this.longitude = 35.1;
+		this.latitude = 0.1;
+		this.arcGisMapServerProvider = MapLayerProviderOptions.ArcGisMapServer;
```
Comment (shiloDrot) [inline demo/app/app.component.ts]: done

File: src/services/basic-primitive-on-map/basic-primitive-on-map.service.ts (hunk 3 above)
Comment (shiloDrot) [inline src/services/basic-primitive-on-map/basic-primitive-on-map.service.ts]: solved

File: demo/app/app.component.ts (hunk 1 above)
Comment (shiloDrot) [inline demo/app/app.component.ts]: solved

File: src/components/ac-html/ac-html.component.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		this.selfElementIsDraw = true;
+	}
+
+	update() {
+		this.remove();
+		this.add();
+	}
+
+	ngOnChanges(changes:SimpleChanges):void {
+		const position = changes['position'];
```
Comment (eitanfr) [inline src/components/ac-html/ac-html.component.ts]: show can also be in the props.
check for changes['show'] and add its logic .

also maybe change `position` to `props` to keep it consist with other components

File: src/components/ac-html/ac-html.component.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+			let screenPosition = Cesium.SceneTransforms.wgs84ToWindowCoordinates(this.cesiumService.getScene(), this.position);
+			this.setScreenPosition(screenPosition);
+		};
+
+		this.cesiumService.getScene().preRender.addEventListener(this.preRenderEventListener);
+		this.selfElementIsDraw = true;
+	}
+
+	update() {
+		this.remove();
```
Comment (eitanfr) [inline src/components/ac-html/ac-html.component.ts]: why need to remove() and add() for update? 
updating the position is not enough?

File: demo/app/layers/draw-on-map-layer/draw-on-map-layer.component.html [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        [props] = "{
+            width: 3,
+            position: position}">
+</ac-point>-->
+<!--<ac-billboard [props] = "{image: '/assets/angry-bird-blue-icon.png',
+                            position: position,
+                            scale: 0.3,
+                            name: 'billboard'}">
+</ac-billboard>-->
+<ac-html #html [position]="position">
```
Comment (eitanfr) [inline demo/app/layers/draw-on-map-layer/draw-on-map-layer.component.html]: also change to props

Comment (shiloDrot) [inline demo/app/layers/draw-on-map-layer/draw-on-map-layer.component.html]: solved

File: src/components/ac-html/ac-html.component.ts (hunk 6 above)
Comment (shiloDrot) [inline src/components/ac-html/ac-html.component.ts]: fixed it

File: src/components/ac-html/ac-html.component.ts (hunk 5 above)
Comment (shiloDrot) [inline src/components/ac-html/ac-html.component.ts]: I added show to props

**Pull request merged**